receiver/windowsperfcountersreceiver/windowsperfcounters_scraper.go

```diff
@@ -145,6 +145,17 @@ func (s *scraper) scrape(context.Context) (pmetric.Metrics, error) {
 			initializeMetricDps(metric, now, val, watcher.MetricRep.Attributes)
 		}
 	}
+
+	// Drop metrics with no datapoints. This happens when configured counters don't exist on the host.
+	// This may result in a Metrics message with no metrics if all counters are missing.
+	metricSlice.RemoveIf(func(m pmetric.Metric) bool {
```

Comment:
is there already a unit test to cover this case?

Reply:
There isn't. I am going to be adding one, but I wanted to make sure the fix was in line with the vision of the receiver (i.e. just confirming that this is indeed a bug and not on purpose).

I'll be pushing the test shortly.